locals.tf

```diff
@@ -33,6 +33,32 @@ locals {
   # Using a one-line command for gpuPostUserData to avoid issues due to different line endings between Windows and Linux.
   gpuPostUserData = "sudo yum -y erase nvidia-driver \nsudo yum -y install make gcc \nsudo yum -y update \nsudo yum -y install gcc kernel-devel-$(uname -r) \nsudo curl -fSsl -O https://us.download.nvidia.com/tesla/${var.gpuNvidiaDriverVersion}/NVIDIA-Linux-x86_64-${var.gpuNvidiaDriverVersion}.run \nsudo chmod +x NVIDIA-Linux-x86_64*.run \nsudo CC=/usr/bin/gcc10-cc ./NVIDIA-Linux-x86_64*.run -s --no-dkms --install-libglvnd \nsudo touch /etc/modprobe.d/nvidia.conf \necho \"options nvidia NVreg_EnableGpuFirmware=0\" | sudo tee --append /etc/modprobe.d/nvidia.conf \nsudo reboot"
 
+  # https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/8a06a6e7006e4bed5630bd49c7434d76c59e0b5e/modules/kubernetes-addons/variables.tf#L183
+  cluster_autoscaler_autodiscovery_tags = [
+    # Helm value array can only be fully replaced, there is no mechanism to just append values to the list of default tags.
+    # Thus, we manually add the default values from
+    #   https://github.com/kubernetes/autoscaler/blob/19fe7aba7ec4007084ccea82221b8a52bac42b34/charts/cluster-autoscaler/values.yaml#L23
+    # here as well:
+    "k8s.io/cluster-autoscaler/enabled",
+    "k8s.io/cluster-autoscaler/${var.infrastructurename}",
+    # and now our additional value(s)
+    # see https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#auto-discovery-setup
+    "k8s.io/cluster-autoscaler/node-template/resources/ephemeral-storage"
+  ]
+  cluster_autoscaler_helm_config = {
+    # NOTE: This version needs to be updated at least on kubernetes version changes (variables.tf: 'kubernetesVersion').
+    #       See https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#releases
+    #       to determine the correct version.
+    version = "9.37.0"

```

Comment:
Maybe a workitem for the backlog: encode EKS version and autoscaler version in a Terraform map, so that the right version is automatically selected.

Reply:
Such a map is officially documented on the linked page. Maintaining such a map in our code would be more effort but no benefit. The map would need to be updated on every version change, too.

What would be better in my opinion is to _scrape_ the corresponding version from some official source ... don't know if this is possible using the linked page. But indeed a separate backlog item and nothing i would do before we have investigated AWS cloud formation as a potential replacement.